src/openrct2-ui/windows/Guest.cpp

```diff
@@ -2222,3 +2335,97 @@ void window_guest_inventory_paint(rct_window* w, rct_drawpixelinfo* dpi)
         gfx_draw_string_left(dpi, STR_NOTHING, nullptr, COLOUR_BLACK, x, y);
     }
 }
+
+void window_guest_debug_resize(rct_window* w)
+{
+    window_set_resize(w, 192 + TabWidth, 159, 500, 450);
+}
+
+void window_guest_debug_invalidate(rct_window* w)
+{
+    if (window_guest_page_widgets[w->page] != w->widgets)
+    {
+        w->widgets = window_guest_page_widgets[w->page];
+        window_init_scroll_widgets(w);
+    }
+
+    w->pressed_widgets |= 1ULL << (w->page + WIDX_TAB_1);
+
+    rct_peep* peep = GET_PEEP(w->number);
+
+    set_format_arg(0, rct_string_id, peep->name_string_idx);
+    set_format_arg(2, uint32_t, peep->id);
+
+    window_guest_debug_widgets[WIDX_BACKGROUND].right = w->width - 1;
+    window_guest_debug_widgets[WIDX_BACKGROUND].bottom = w->height - 1;
+
+    window_guest_debug_widgets[WIDX_PAGE_BACKGROUND].right = w->width - 1;
+    window_guest_debug_widgets[WIDX_PAGE_BACKGROUND].bottom = w->height - 1;
+
+    window_guest_debug_widgets[WIDX_TITLE].right = w->width - 2;
+
+    window_guest_debug_widgets[WIDX_CLOSE].left = w->width - 13;
+    window_guest_debug_widgets[WIDX_CLOSE].right = w->width - 3;
+
+    window_align_tabs(w, WIDX_TAB_1, WIDX_TAB_7);
+}
+
+void window_guest_debug_paint(rct_window* w, rct_drawpixelinfo* dpi)
+{
+    window_draw_widgets(w, dpi);
+    window_guest_overview_tab_paint(w, dpi);
+    window_guest_stats_tab_paint(w, dpi);
+    window_guest_rides_tab_paint(w, dpi);
+    window_guest_finance_tab_paint(w, dpi);
+    window_guest_thoughts_tab_paint(w, dpi);
+    window_guest_inventory_tab_paint(w, dpi);
+    window_guest_debug_tab_paint(w, dpi);
+
+    rct_peep* peep = GET_PEEP(w->number);
+
+    // cx
+    int32_t x = w->x + window_guest_debug_widgets[WIDX_PAGE_BACKGROUND].left + 4;
+    // dx
+    int32_t y = w->y + window_guest_debug_widgets[WIDX_PAGE_BACKGROUND].top + 4;
+
+    char buffer[512];
+    snprintf(buffer, sizeof(buffer), "Position: %i, %i, %i", peep->x, peep->y, peep->z);
+    gfx_draw_string(dpi, buffer, COLOUR_BLACK, x, y);
+    y += LIST_ROW_HEIGHT;
+
+    snprintf(buffer, sizeof(buffer), "Next: %u, %u, %u", peep->next_x, peep->next_y, peep->next_z);
```

Comment:
Are these actually useful? next_x and x are usually the same. I can't quite remember the exact design of them.